packages/go/analysis/azure/azrole/approver.go

```diff
@@ -0,0 +1,150 @@
+				// Step 2: isApprovalRequired == true
+				query.Equals(
+					query.NodeProperty(azure.EndUserAssignmentRequiresApproval.String()),
+					true,
+				),
+				// Step 2: primaryApprovers (user or group) is not null
+				query.Or(
+					query.IsNotNull(
+						query.NodeProperty(azure.EndUserAssignmentUserApprovers.String()),
+					),
+					query.IsNotNull(
+						query.NodeProperty(azure.EndUserAssignmentGroupApprovers.String()),
+					),
+				),
+			)
+		}))
+		if err != nil {
+			return err
+		}
+		fetchedAZRoles = fetchedNodes
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	for _, fetchedAZRole := range fetchedAZRoles {
+		if err := operation.Operation.SubmitReader(func(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob) error {
+			// Step 3a: Read the primaryApprovers list (group GUIDs)
+			userApproversID, err := fetchedAZRole.Properties.Get(
+				azure.EndUserAssignmentUserApprovers.String(),
+			).StringSlice()
+			if err != nil {
+				return err
+			}
+			groupApproversID, err := fetchedAZRole.Properties.Get(
+				azure.EndUserAssignmentGroupApprovers.String(),
+			).StringSlice()
+			if err != nil {
+				return err
+			}
+			principalIDs := append(userApproversID, groupApproversID...)
+			if len(principalIDs) == 0 {
+				// Handle default admin roles...
+				return handleDefaultAdminRoles(ctx, tx, outC, tenantID, fetchedAZRole)
+			} else {
+				// Handle principal IDs...
+				return handlePrincipalApprovers(ctx, tx, outC, principalIDs, fetchedAZRole)
+			}
+		}); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func handleDefaultAdminRoles(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob, principalID string, fetchedAZRole *graph.Node) error {
+	fetchedNodes, err := ops.FetchNodeSet(tx.Nodes().Filterf(func() graph.Criteria {
+		return query.And(
+			query.Equals(query.NodeProperty(azure.TenantID.String()), principalID),
+			query.Kind(query.Node(), azure.Role),
+			query.Or(
+				query.Kind(query.Node(), azure.GlobalAdmin),
+				query.Kind(query.Node(), azure.PrivilegedRoleAdmin),
+			),
+		)
+	}))
+	if err != nil {
+		return err
+	}
+
+	for _, fetchedNode := range fetchedNodes {
+		// enqueue creation of AZRoleApprover edge: from fetchedNode → fetchedAZRole
+		channels.Submit(ctx, outC, analysis.CreatePostRelationshipJob{
+			FromID: fetchedNode.ID,
+			ToID:   fetchedAZRole.ID,
+			Kind:   azure.AZRoleApprover,
+		})
+	}
+
+	return nil
+}
+
+func handlePrincipalApprovers(ctx context.Context, tx graph.Transaction, outC chan<- analysis.CreatePostRelationshipJob, principalIDs []string, fetchedAZRole *graph.Node) error {
+	for _, principalID := range principalIDs {
+		fetchedNode, err := tx.Nodes().Filterf(func() graph.Criteria {
+			return query.And(
+				query.Kind(query.Node(), azure.Entity),
+				query.Equals(query.NodeProperty(common.ObjectID.String()), principalID),
+			)
+		}).First()
+
+		var nodeID graph.ID
+		if err != nil {
```

Comment:
This can be cleaned up:

```
		var nodeID graph.ID
		if err != nil {
			if graph.IsErrNotFound(err) {
				slog.WarnContext(ctx, fmt.Sprintf("Entity node not found for principal ID: %s, skipping edge creation", principalID))
				continue
			}

			return err
		}

		nodeID = fetchedNode.ID
```